patroni/postgresql.py

```diff
@@ -564,13 +560,12 @@ def cleanup_archive_status(self):
     def follow(self, leader, recovery=False):
         if self.check_recovery_conf(leader) and not recovery:
             return True
-
         change_role = self.role == 'master'
```

Comment:
Unfortunately it will not fix https://github.com/zalando/patroni/issues/167
Because role would be 'replica' there